### PR TITLE
Migrate from dependabot to renovate

### LINKED ISSUE #5115: Move to Renovate
We recently ran into a couple issues related to dependabot plus the already known shortcomings of and hacks for dependabot.

Recent issues:
* Dependency updates run for `projectnessie/nessie` took 7+ hours, I got a "solid" answer from GH support that it's caused by a 1.5M `package-lock.json` (not really helpful)

Known dependabot shortcomings:
* Dependabot's support for github-actions does not work - error messages are not helpful at all (see #5110)
* Dependabot's support for Gradle is insufficient and requires a couple of hacks in the build scripts (see `gradle/dependabot/build.gradle.kts` and hack-y code for that)
* Many people complain about those and other dependabot issues and move to Renovate

Trying out [Renovate](https://www.mend.io/free-developer-tools/renovate/) ([Docs here](https://docs.renovatebot.com/)) was straightforward, very easy and without any issues. Exchanged dependabot w/ Renovate on my own forks of Nessie, CEL-Java, Gradle-Build-Plugins and NesQuEIT - works like a charm - those forks currently use Renovate.

Renovate has quite a few interesting features and relevant improvements over dependabot:
* Seamlessly supports github-actions - the update of the version relates to the "version type being used": means, if you use e.g. `v3`, it updates to `v4`, if you use `v3.1` it updates to `v3.2` and so on.
* Supports Gradle wrapper updates
* Supports "pacakge groups" - i.e. updates related packages in a single PR (dependabot is famous for creating different PRs in those situations - and hence requires manual changes)
* Automerge can be configured for major/minor/patch/pin/digest version type updates, manager (maven/gradle/github-actions/pip/etc) and package
* Generally, Renovate can be configured in many ways - and it supports JSON5 (JSON for humans, incl comments)
* Nonetheless Renovate runs on a variety of platforms (github, gitlab, on-prem, locally w/ CLI, many more)
* Very informative PR descriptions and comments + good logs
* Free for OSS
* Oh - and forks do not run Renovate unless you explicitly ask it to do so

The only disadvantage is that it takes a bit longer to automatically merge the dependency-update PRs.

### PR DESCRIPTION
**Pull request by snazy**
Adds the renovate configuration, removes the dependabot configuration and some hacks for dependabot.

Non-major version updates have automerge enabled for managers that have CI coverage: maven, gradle, gradle-wrapper, npm, pip*.

Other managers for which renovatebot generates PRs: github actions, helm.

Also includes rules to not perform major version bumps for some well known major-version-upgrade issues.

This PR enables the migration to Gradle version catalogs, which will eliminate all remaining dependabot hacks. That change, which builds on this one, is available for preview [here](https://github.com/snazy/nessie/tree/version-catalogs). It will nearly completely remove the version declarations in `settings.gradle.kts` and `nessie-iceberg/settings.gradle.kts`.

This PR is currently WIP. It's already running for a while on [my Nessie fork](https://github.com/snazy/nessie) and for [cel-java](https://github.com/snazy/cel-java), [gradle-build-plugins](https://github.com/snazy/gradle-build-plugins) and [NesQuEIT]([query-engine-integration-tests](https://github.com/snazy/query-engine-integration-tests) without any issues.

TODO (admin tasks to get this change merged):
* Add renovate-approve + Renovate GitHub apps
  * https://github.com/apps/renovate-approve
  * https://github.com/apps/renovate
* Add `renovate-approve` app user to allowed approvers
* Update branch protection rules

Follow-up #5167 removes the remainders of the band-aids that were added only for dependabot.

Fixes #5115

### CODE REVIEW

File: .github/workflows/release-publish.yml [hunk 1]
```diff
@@ -216,7 +216,7 @@ jobs:
         LAST_TAG=$(git describe --abbrev=0 --tags --match=nessie-* ${GIT_TAG}^1)
         NUM_COMMITS=$(git log --format='format:%h' ${LAST_TAG}..HEAD^1 | wc -l)
 
-        git log --perl-regexp --author '^(?!.*dependabot(-preview)?|.*nessie-release-workflow).*$' --format='format:* %s' ${LAST_TAG}..${GIT_TAG} | grep -v '^\* \[release\] .*$' > ${DIR}/release-log
+        git log --perl-regexp --author '^(?!.*renovate|.*dependabot(-preview)?|.*nessie-release-workflow).*$' --format='format:* %s' ${LAST_TAG}..${GIT_TAG} | grep -v '^\* \[release\] .*$' > ${DIR}/release-log
```
Comment (XN137) [inline .github/workflows/release-publish.yml:219]: due to your recent PRs i think this adjustment needs to be done in more places now...

generally we should grep the repo for "dependabot" and make sure all remaining usage spots are necessary

File: .github/renovate.json5 [hunk 2]
```diff
@@ -0,0 +1,54 @@
+{
+  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
+  extends: [
+    "config:base",
+    // https://docs.renovatebot.com/presets-npm/
+    "npm:unpublishSafe",
+    // Required to not pin dependencies to _exact_ versions (npm + pip)
+    ":preserveSemverRanges"
```
Comment (XN137) [inline .github/renovate.json5:8]: tiny nit: i think json5 supports trailing commas (in both arrays and objects)

**Review by XN137 — COMMENTED** [review]
changes look good to me but a few changes in recently added release workflows are missing i think.
also PR description still mentions this is a draft
otherwise +1

personally i would be fine with manual approval+merging in order to not give write access to the apps, but i guess the larger group needs to decide if thats a problem or not.
either way looking forward to replacing dependabot :+1:

**Comment by snazy** [comment]
Yea, intentionally still a draft (to not accidentally merge it)

**Comment by snazy** [comment]
> few changes in recently added release workflows are missing i think

which ones?

**Comment by snazy** [comment]
Auto-approval/merge:
The change allows automerge for miinor/patch version changes, but not for major version bumps.

File: nessie-iceberg/settings.gradle.kts [hunk 3]
```diff
@@ -40,56 +40,45 @@ pluginManagement {
...
   }
 
-  plugins {
+  val versionErrorPronePlugin = "2.0.2"
+  val versionIdeaExtPlugin = "1.1.6"
+  val versionJandexPlugin = "1.82"
+  val versionNessiePlugins = "0.2.12"
+  val versionQuarkus = "2.13.0.Final"
+  val versionShadowPlugin = "7.1.2"
+  val versionSpotlessPlugin = "6.11.0"
```
Comment (XN137) [inline nessie-iceberg/settings.gradle.kts:49]: is there no way for these to reuse version mapping from root settings?

**Review by XN137 — DISMISSED** [review]

File: nessie-iceberg/settings.gradle.kts (hunk 3 above)
Comment (snazy) [reply nessie-iceberg/settings.gradle.kts:49]: Those will be managed externally via the version-catalog PR

Comment (snazy) [reply nessie-iceberg/settings.gradle.kts:49]: Eventually get rid of those entirely at some time

**Comment by snazy** [comment]
Last commit should fix a scheduling issue. Minor-patch updates for pip+npm were correctly scheduled on weekends, but major pip+npm updates ran on the default schedule (any time).

**Review by XN137 — APPROVED** [review]

**Pull request merged**